### PR TITLE
More intelligent build arch default in macos scheme for OpenSSL

### PR DESCRIPTION
**Pull request by TolonUK**
The MacOS scheme for OpenSSL defaults to "x86_64" if CMAKE_OSX_ARCHITECTURES is not set. This can result in a broken build on M1 (arm64) Macs if the CMake configuration depends on OpenSSL via Hunter. This pull request replaces the "x86_64" default with CMAKE_SYSTEM_PROCESSOR.

* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **[Yes]**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **[Yes]**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **[Yes]**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **[Yes]**

### CODE REVIEW

**Comment by NeroBurner** [comment]
@TolonUK Do you by any chance know if there is a macos M1 runner available on github? If so would you be interested to add it to our build matrix?

**Comment by TolonUK** [comment]
Still no macos M1 runner available on GitHub, see https://github.com/actions/runner/issues/805. I'm happy to take a look at running the tests locally, but I'll need to figure out how to do that first.

Will the lack of CI testing block this PR?

**Comment by NeroBurner** [comment]
The lack of GitHub provided runner won't block this PR, thanks for the info and link about it

I'll merge if you have tested the changes locally to make sure the change works on M1

The easiest way to do that locally (to my knowledge) is to build the corresponding example project inside the Hunter repository. You have to have the `gate` submodule available:
```sh
git submodule update --init gate
cd examples/OpenSSL
cmake -S . -B build
cmake --build build
```

**Comment by TolonUK** [comment]
I've tested the change on an M1 Mac with the project that I originally had the issue with - all the libraries were correctly building as arm64, except for OpenSSL. This change resolved my issue.

Is that enough, or would you like me to build the OpenSSL example as well?

**Review by NeroBurner — APPROVED** [review]

**Comment by NeroBurner** [comment]
That should be enough, thanks. Out of curiosity, how did you build your project against your fixed Hunter build

**Comment by TolonUK** [comment]
I cleared my caches and used the commit sha on my fork:

```
...
include(cmake/HunterGate.cmake)
HunterGate(
    URL "https://github.com/TolonUK/hunter/archive/6438890eb5b0fd7c659acc962bb0387d9b8d1a71.tar.gz"
    SHA1 "b0bd98524c754d36939af51b1ad4aef60fe8c054"
    LOCAL
)
...
```

**Comment by NeroBurner** [comment]
Ah yes, good solution. Another one for local development would be to set the cmake variable `HUNTER_ROOT` to a local folder with the checked out hunter repository.

Github doesn't seem to like me currently and won't enable the merge button. I'll fix that and merge this PR as soon as possible

**Pull request merged**
